### ISSUE TITLE
Error with model.evaluate(): Error when checking target: expected dense_16 to have shape (7,) but got array with shape (1,)

### ISSUE THREAD

**Issue by benn510** [keras-team/keras#12410]
Hi everyone, I'm new on keras and i'm developing a Neural Network model for my thesis. 
This is a credit scoring model and the network here should assign the right rating for a client.
The output goes from 0 to 6.

Here's my code:
```
`  model = Sequential()
model.add(Dense(units = 90, kernel_initializer='uniform', activation = 'relu', input_dim=len(d2.dtypes)-1))
model.add(Dense(units = 130, kernel_initializer='uniform', activation = 'relu'))
model.add(Dense(units = 150, kernel_initializer='uniform', activation = 'relu'))
model.add(Dense(units = 7, kernel_initializer='uniform', activation='softmax'))

adam = optimizers.Adam()

model.compile(loss='categorical_crossentropy', metrics = ['accuracy'], optimizer = adam)

y = d2['grade']
X = d2.drop(['grade'], axis=1)

X_train, X_test, Y_train, Y_test = train_test_split(X,y, test_size=0.2, shuffle=True)

one_hot_labels = keras.utils.to_categorical(Y_train, num_classes=7)

model.fit(X_train, one_hot_labels, batch_size=32, epochs=20, validation_split=0.2, callbacks=[TensorBoardColabCallback(tboard)])

one_hot_y_test = keras.utils.to_categorical(Y_test, num_classes=7)

# Predicting the Test set results
y_pred = model.predict(X_test)
y_pred = (y_pred > 0.5)
```

```
x = pd.DataFrame(columns=['0','1','2','3','4','5','6'], data=y_pred)
p = x.idxmax(1)
p = p.astype(int)
```

```
# Creating the Confusion Matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(Y_test, p)
cm `
```

But when I try to compute the test accuracy, it gives me the following error:

```
acc= model.evaluate(X_test, Y_test, verbose=0)
print('Test accuracy:', acc)
```
Error when checking target: expected dense_16 to have shape (7,) but got array with shape (1,)

Could someone kinda help me, please?

**Comment by sathk882** [comment]
Hi,

I think in the evaluate method you are giving wrong target values. You need to send "one_hot_y_test" instead of "Y_test".

**Comment by thedivyanshuyadav** [comment]
@benn510  Error #12410 should be close now as @satheeshkatipomu has answered the fix for this issue.
